### ISSUE TITLE
Different YOLO work on the same video with different bitrate and weight

### ISSUE THREAD

**Issue by Egorundel** [ultralytics/ultralytics#18858]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Greetings.

It turned out that my video objects are poorly detected, which have a high bitrate and weigh a lot of MB:

![Image](https://github.com/user-attachments/assets/653dac94-76a8-4d7b-adcf-2aed132f1e74)

In the first case, the video weighs 146.2 KB and a bitrate of 155 kbps, and in the second 248.1 MB and a bitrate of 271120 kbps. It's the same video, it just weighs differently. And detection works better for the first case (low weight and low bitrate).

What could be the problem? 

P.S. I'm working with YOLOv11 on TensorRT for C++.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Egorundel, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for YOLO-related insights, where you can explore various [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions are addressed there, and it’s a great resource to troubleshoot issues.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug the problem. It would be particularly helpful to have details of how the video data was processed, specific configuration settings, and the TensorRT inference pipeline.

If this is a ❓ Question, could you please share additional details about your workflow? For example:
- How were the high and low bitrate versions of the video prepared?
- What preprocessing steps (if any) were applied to these videos before inference?
- Are there notable differences in detection confidence, FPS, or output logs?

Meanwhile, make sure you are following our guidelines to maximize detection performance. The [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) section may also offer some insights.

## Upgrade

Please ensure you're using the latest `ultralytics` package to rule out any resolved issues. You can upgrade as follows in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO models can run in a variety of setups. Here are some verified environments:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image** for reproducibility: See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

You can verify the system-wide tests for YOLO by checking our CI badge below:

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

This badge will show the status of our [Continuous Integration](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule), which tests all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) daily across macOS, Windows, and Ubuntu.

Also, this is an automated response to your issue, but don’t worry—a dedicated Ultralytics engineer will review your report and provide further insights. Let us know if you can offer more context in the meantime! 😊

**Comment by Y-T-G** [comment]
> I'm working with YOLOv11 on TensorRT for C++.

What's the results when you use Ultralytics?

**Comment by Egorundel** [comment]
@Y-T-G When I use Python and `weights.pt`?

**Comment by Egorundel** [comment]
When I use your pre-trained YOLOv11(m) model and Python code:

```python
from ultralytics import YOLO

# Load a pretrained YOLO11n model
model = YOLO("yolo11m.pt")

# Run inference on 'bus.jpg' with arguments
# model.predict("./video0.mp4", save=True, imgsz=640, conf=0.4)
model.predict("./video1.mp4", save=True, imgsz=640, conf=0.4)
```

Then I have different detection results.

**Comment by Y-T-G** [comment]
The detection would be different because the quality is different.

**Comment by Egorundel** [comment]
@Y-T-G Thanks for your reply. But I'll ask the question a little differently.

The problem is that the detection works better on a low-bitrate video than on that high-bitrate video, although I train on **uncompressed** images. 

Maybe when training YOLOv11, she compresses them there herself to improve the learning rate? If there is any parameter for image compression during training, then I would like to disable it.

**Comment by Y-T-G** [comment]
Did you use default augmentations?

**Comment by Egorundel** [comment]
@Y-T-G 

> Did you use default augmentations?

I'm not changing anything about the hyperparameters regarding augmentation.

I started the training using this configuration:
```python
from ultralytics import YOLO
 
# Load the model.
model = YOLO('yolo11m.pt')

model.train(
   data = './experiment/coco_customYOLOv11_NEW.yaml',
   classes=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
   epochs=100,
   warmup_epochs=0,
   optimizer="SGD",
   lr0=0.0005,
   batch=32,
   imgsz=640,
   device=(0),
   name='yolov11_custom_night',
   project='./experiment/trains_YOLOv11_night/YOLOv11_m_night_640'
)
```

**Comment by Egorundel** [comment]
@Y-T-G 

![Image](https://github.com/user-attachments/assets/bcc32dc7-5a28-4a6b-9305-8c0e750e1b1f)

I just noticed that my augmentation is disabled. Do I need to turn it on?

**Comment by Y-T-G** [comment]
It's enabled by default. That's a different parameter.

**Comment by Egorundel** [comment]
@Y-T-G Can you give me any advice in connection with my problem described above?

**Comment by Y-T-G** [comment]
Does the issue exist when you use your model with Ultralytics?

**Comment by Egorundel** [comment]
@Y-T-G yes

**Comment by Egorundel** [comment]
@Y-T-G What hyperparameters should I set so that the model learns better and then works on images/videos without compression?

**Comment by Egorundel** [comment]
This hyperparameter `multi_scale` did not help me to overcome my problem.

![Image](https://github.com/user-attachments/assets/c38189cb-3206-40f0-926d-987bfc3f0053)

**Comment by Y-T-G** [comment]
You can try this:

https://community.ultralytics.com/t/random-cropping-as-an-augmentation/546/10

**Comment by Egorundel** [comment]
@Y-T-G 

> You can try this:
>
> https://community.ultralytics.com/t/random-cropping-as-an-augmentation/546/10

Thank you, should I start training from scratch or can I train a pretrained model using your repository?

**Comment by Y-T-G** [comment]
You can do either

**Comment by Egorundel** [comment]
@Y-T-G Okay, thanks, I'll try to train an already pre-trained model.

![Image](https://github.com/user-attachments/assets/95409e5a-f90e-489f-a1ea-d0db4defd3f0)

**Comment by Egorundel** [comment]
@Y-T-G I am training a pre-trained model with `SGD`, `no warmup_epochs`, `lr 0.0005` and `hi_res=True` parameters, are the results good in the first 10 epochs?

![Image](https://github.com/user-attachments/assets/f612fb7a-6a3e-43ef-99d0-260d7941d9ea)
